packages/react/navigation-menu/src/NavigationMenu.tsx

```diff
@@ -504,13 +506,22 @@ const NavigationMenuLink = React.forwardRef<NavigationMenuLinkElement, Navigatio
           aria-current={active ? 'page' : undefined}
           {...linkProps}
           ref={forwardedRef}
-          onClick={composeEventHandlers(props.onClick, (event) => {
-            const rootContentDismissEvent = new Event(ROOT_CONTENT_DISMISS, {
-              bubbles: true,
-              cancelable: true,
-            });
-            event.target.dispatchEvent(rootContentDismissEvent);
-          })}
+          onClick={(event) => {
+            props.onClick?.(event);
+
+            const target = event.target;
+            const linkSelectEvent = new Event(LINK_SELECT, { bubbles: true, cancelable: true });
+            target.addEventListener(LINK_SELECT, (event) => onSelect?.(event), { once: true });
+            target.dispatchEvent(linkSelectEvent);
```

Comment:
Felt clearer to use an additional event vs piggybacking `rootContentDismissEvent`

Reply:
It makes me wonder if it should just be an `onDismiss` custom event then given that the linking behaviour wouldn't be attached to `onSelect` like it is on menus?

On menus, `onSelect` makes sense, because people attach strictly onto this for behaviour.

Although, I like the consistency, not sure which way around it should be…

Reply:
Caught up offline, decided that although it doesn’t feel great to have to provide this in the context of a `Link`, `onSelect` does map neatly with existing DX and the user interaction itself.